### PR TITLE
KFSPTS-19974: Update to explicitly use Locale.US for formatting operations.

### PR DESCRIPTION
**Pull request by nkimble**

### CODE REVIEW

**Comment by nkimble** [comment]
This PR is ready for code review but should not be merged until we coordinate an appropriate time with the functional users.

**Comment by nkimble** [comment]
Reopening this PR to have files reviewed with any issues addressed now.
Please do not merge this PR until after the 12/17 upgrade patch release is merged to develop first.

File: src/main/java/edu/cornell/kfs/sys/xmladapters/StringToJavaDateAdapter.java [hunk 1]
```diff
@@ -16,7 +17,7 @@
  */
 public class StringToJavaDateAdapter extends XmlAdapter<String, Date> {
 
-    private static final DateTimeFormatter DATE_FORMATTER = DateTimeFormat.forPattern("MM/dd/yyyy");
+    private static final DateTimeFormatter DATE_FORMATTER = DateTimeFormat.forPattern("MM/dd/yyyy").withLocale(Locale.US);
```
Comment (jhulslander) [inline src/main/java/edu/cornell/kfs/sys/xmladapters/StringToJavaDateAdapter.java:20]: can the format string be moved to a new constant in CUKFSConstants

File: src/test/java/edu/cornell/kfs/fp/batch/service/impl/CreateAccountingDocumentServiceImplTest.java [hunk 2]
```diff
@@ -142,7 +143,7 @@
     private static final String TARGET_TEST_FILE_PATH = "test/fp/accountingXmlDocument";
     private static final String FULL_FILE_PATH_FORMAT = "%s/%s%s";
     private static final int DOCUMENT_NUMBER_START = 1000;
-    private static final SimpleDateFormat DATE_FORMAT = new SimpleDateFormat("yyyyMMdd-HH-mm-ss-S");
+    private static final SimpleDateFormat DATE_FORMAT = new SimpleDateFormat("yyyyMMdd-HH-mm-ss-S", Locale.US);
```
Comment (jhulslander) [inline src/test/java/edu/cornell/kfs/fp/batch/service/impl/CreateAccountingDocumentServiceImplTest.java:146]: can the format string be moved to a new constant in CUKFSConstants

File: src/test/java/edu/cornell/kfs/pdp/batch/service/impl/CuExtractPaymentServiceImplTest.java [hunk 3]
```diff
@@ -43,8 +45,8 @@
     private static final String ACH_EXTRACT_TEST_FILE_NAME = "testAchExtractPaymentsOuput.txt";
     private static final String ACH_EXTRACT_TEST_PATH_AND_FILE = ACH_EXTRACT_TEST_FILE_PATH + File.separator + ACH_EXTRACT_TEST_FILE_NAME;
     private static final String ACH_EXTRACT_TEST_PATH_AND_FILE_FOR_VALIDATION = ACH_EXTRACT_TEST_PATH_AND_FILE + ".READY";
-    private static final SimpleDateFormat yyyyMMddHHmmss_DATE_FORMATTER = new SimpleDateFormat("yyyyMMddHHmmss");
-    private static final SimpleDateFormat yyyyMMdd_DATE_FORMATTER = new SimpleDateFormat("yyyyMMdd");
+    private static final SimpleDateFormat yyyyMMddHHmmss_DATE_FORMATTER = new SimpleDateFormat("yyyyMMddHHmmss", Locale.US);
```
Comment (jhulslander) [inline src/test/java/edu/cornell/kfs/pdp/batch/service/impl/CuExtractPaymentServiceImplTest.java:48]: can the format string be moved to a new constant in CUKFSConstants

**Review by jhulslander — CHANGES_REQUESTED** [review]
Changes look very good overall, I found some commented out variables I think can be removed, and a few cases where some format strings could be moved to new constants in CUKFSConstants

File: src/main/java/com/rsmart/kuali/kfs/cr/businessobject/CheckReconciliationReport.java [hunk 4]
```diff
@@ -38,11 +42,11 @@
     private String status;
     private String amount;
     private Double subTotal;
-    public static SimpleDateFormat MONYYF = new SimpleDateFormat("yyyy/MM");
-    public static SimpleDateFormat SDF = new SimpleDateFormat("MM/dd/yyyy");
-    public static DecimalFormat DF = new DecimalFormat("#0.00");
+    public static SimpleDateFormat MONYYF = new SimpleDateFormat("yyyy/MM", Locale.US);
+    public static SimpleDateFormat SDF = new SimpleDateFormat(KFSConstants.MONTH_DAY_YEAR_DATE_FORMAT, Locale.US);
+    public static DecimalFormat DF = new DecimalFormat("#0.00", new DecimalFormatSymbols(Locale.US));
```
Comment (nkimble) [inline src/main/java/com/rsmart/kuali/kfs/cr/businessobject/CheckReconciliationReport.java:48]: Added a constant for this string as well.

File: src/main/java/com/rsmart/kuali/kfs/fp/batch/DisbursementVoucherInputFileType.java [hunk 5]
```diff
@@ -45,7 +46,7 @@ public String getFileName(String principalId, Object parsedFileContents, String
         Timestamp currentTimestamp = dateTimeService.getCurrentTimestamp();
 
         StringBuffer buf = new StringBuffer();
-        SimpleDateFormat formatter = new SimpleDateFormat("yyyyMMdd_HHmmss");
+        SimpleDateFormat formatter = new SimpleDateFormat("yyyyMMdd_HHmmss", Locale.US);
```
[the remark this answers is not in the document]
Comment (nkimble) [reply src/main/java/com/rsmart/kuali/kfs/fp/batch/DisbursementVoucherInputFileType.java:49]: Added constant

File: src/main/java/edu/cornell/kfs/fp/batch/service/impl/AdvanceDepositServiceImpl.java [hunk 6]
```diff
@@ -287,7 +289,7 @@ protected AdvanceDepositDocument createAdvanceDepositDocument(AchIncomeTransacti
     }
 
     private void createNotes(AchIncomeTransaction transaction, AdvanceDepositDocument document) {
-        String fileName = CuFPConstants.ADVANCE_DEPOSIT_NOTE_FILE_PREFIX + document.getDocumentNumber() + "_" + new SimpleDateFormat("yyyyMMdd_HHmmss").format(dateTimeService.getCurrentDate()) + ".txt";
+        String fileName = CuFPConstants.ADVANCE_DEPOSIT_NOTE_FILE_PREFIX + document.getDocumentNumber() + "_" + new SimpleDateFormat("yyyyMMdd_HHmmss", Locale.US).format(dateTimeService.getCurrentDate()) + ".txt";
```
[the remark this answers is not in the document]
Comment (nkimble) [reply src/main/java/edu/cornell/kfs/fp/batch/service/impl/AdvanceDepositServiceImpl.java:292]: Constant used for the format string.

File: src/main/java/edu/cornell/kfs/module/purap/batch/IWantDocumentInputFileType.java [hunk 7]
```diff
@@ -23,7 +24,7 @@ public String getFileName(String principalId, Object parsedFileContents, String
         Timestamp currentTimestamp = dateTimeService.getCurrentTimestamp();
 
         StringBuffer buf = new StringBuffer();
-        SimpleDateFormat formatter = new SimpleDateFormat("yyyyMMdd_HHmmss");
+        SimpleDateFormat formatter = new SimpleDateFormat("yyyyMMdd_HHmmss", Locale.US);
```
[the remark this answers is not in the document]
Comment (nkimble) [reply src/main/java/edu/cornell/kfs/module/purap/batch/IWantDocumentInputFileType.java:27]: Changed to use a constant.

File: src/main/java/edu/cornell/kfs/module/receiptProcessing/service/impl/ReceiptProcessingServiceImpl.java [hunk 8]
```diff
@@ -486,7 +487,7 @@ protected void matchOrAttachOnly(String fileName, BatchInputFileType batchInputF
     protected void getcsvWriter(String csvDoc, String reportDropFolder) {
         
         String fileName = "CIT_OUT_" +
-            new SimpleDateFormat("yyyyMMdd_HHmmssSSS").format(dateTimeService.getCurrentDate()) + ".csv";
+            new SimpleDateFormat("yyyyMMdd_HHmmssSSS", Locale.US).format(dateTimeService.getCurrentDate()) + ".csv";
```
[the remark this answers is not in the document]
Comment (nkimble) [reply src/main/java/edu/cornell/kfs/module/receiptProcessing/service/impl/ReceiptProcessingServiceImpl.java:490]: Constant created and used.

File: src/main/java/edu/cornell/kfs/sys/batch/CuAbstractStep.java [hunk 9]
```diff
@@ -39,7 +40,7 @@ public void addTimeStampToFileName(File origFile, String fileName, String direct
 		String fileNameProper = fileName.substring(0, fileName.lastIndexOf('.'));
 		String fileExtension = fileName.substring(fileName.lastIndexOf('.'));
 
-		DateFormat df = new SimpleDateFormat("MMddyyyy_hhmmss");
+		DateFormat df = new SimpleDateFormat("MMddyyyy_hhmmss", Locale.US);
```
[the remark this answers is not in the document]
Comment (nkimble) [reply src/main/java/edu/cornell/kfs/sys/batch/CuAbstractStep.java:43]: Constant created and used.

File: src/main/java/edu/cornell/kfs/sys/xmladapters/StringToJavaDateAdapter.java (hunk 1 above)
Comment (nkimble) [reply src/main/java/edu/cornell/kfs/sys/xmladapters/StringToJavaDateAdapter.java:20]: Constant created.

File: src/test/java/edu/cornell/kfs/fp/batch/service/impl/CreateAccountingDocumentServiceImplTest.java (hunk 2 above)
Comment (nkimble) [reply src/test/java/edu/cornell/kfs/fp/batch/service/impl/CreateAccountingDocumentServiceImplTest.java:146]: Constant created and used.

File: src/test/java/edu/cornell/kfs/pdp/batch/service/impl/CuExtractPaymentServiceImplTest.java (hunk 3 above)
Comment (nkimble) [reply src/test/java/edu/cornell/kfs/pdp/batch/service/impl/CuExtractPaymentServiceImplTest.java:48]: Constant used.

**Comment by nkimble** [comment]
Unit tests are running. This is ready to be reviewed again.

File: src/main/java/edu/cornell/kfs/module/ld/batch/service/impl/CuFileEnterpriseFeederHelperServiceImpl.java [hunk 10]
```diff
@@ -106,7 +107,7 @@ public void feedOnFile(File doneFile, File dataFile, File reconFile, PrintStream
                     
                 Collection<String> offsetDocTypes = parameterService.getParameterValuesAsString(
                         LaborEnterpriseFeedStep.class, LaborParameterConstants.LABOR_BENEFIT_OFFSET_DOCTYPE);
-                offsetDocTypes = offsetDocTypes.stream().map(String::toUpperCase).collect(Collectors.toList());
+                offsetDocTypes = offsetDocTypes.parallelStream().map(offsetDocType -> offsetDocType.toUpperCase(Locale.US)).collect(Collectors.toList());
```
Comment (jhulslander) [inline src/main/java/edu/cornell/kfs/module/ld/batch/service/impl/CuFileEnterpriseFeederHelperServiceImpl.java:110]: Did you intentially switch from stream to parallelStream?

File: src/main/java/edu/cornell/kfs/sys/CUKFSConstants.java [hunk 11]
```diff
@@ -22,6 +22,15 @@
     public static final String DATE_FORMAT_mm_dd_yyyy_hh_mm_ss_am = "MM/dd/yyyy hh:mm:ss a";
     public static final String DATE_FORMAT_yyyy_MM_dd_T_HH_mm_ss_SSS = "yyyy-MM-dd'T'HH:mm:ss.SSS";
     public static final String DATE_FORMAT_dd_MMM_yy = "dd-MMM-yy";
+    public static final String DATE_FORMAT_yyyyMMddHHmmss = "yyyyMMddHHmmss";
+    public static final String DATE_FORMAT_yyyy_MM = "yyyy/MM";
+    public static final String DATE_FORMAT_yyyyMMdd_HHmmss = "yyyyMMdd_HHmmss";
+    public static final String DATE_FORMAT_yyyyMMdd_HHmmssSSS = "yyyyMMdd_HHmmssSSS";
+    public static final String DATE_FORMAT_MMddyyyy_hhmmss = "MMddyyyy_hhmmss";
+    public static final String DATE_FORMAT_MM_dd_yyyy = "MM/dd/yyyy";
```
Comment (jhulslander) [inline src/main/java/edu/cornell/kfs/sys/CUKFSConstants.java:30]: I think this is the same as KFSConstants.MONTH_DAY_YEAR_DATE_FORMAT

**Review by jhulslander — CHANGES_REQUESTED** [review]
Things are looking good, I had just a couple of questions.

File: src/main/java/edu/cornell/kfs/sys/CUKFSConstants.java (hunk 11 above)
Comment (nkimble) [reply src/main/java/edu/cornell/kfs/sys/CUKFSConstants.java:30]: Yes it is. Adjusting for that change.

File: src/main/java/edu/cornell/kfs/module/ld/batch/service/impl/CuFileEnterpriseFeederHelperServiceImpl.java (hunk 10 above)
Comment (nkimble) [reply src/main/java/edu/cornell/kfs/module/ld/batch/service/impl/CuFileEnterpriseFeederHelperServiceImpl.java:110]: Thanks for catching.

**Comment by jhulslander** [comment]
retest this please

**Review by jhulslander — APPROVED** [review]
my concerns have been addressed, unit tests pass, I shall merge

**Pull request merged**